packages/p2p/src/components/modal-manager/modals/quick-add-modal/quick-add-modal.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import classNames from 'classnames';
+import { DesktopWrapper, Icon, MobileFullPageModal, MobileWrapper, Modal } from '@deriv/components';
+import { observer } from '@deriv/stores';
+import { useStores } from 'Stores';
+import { localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { buy_sell } from 'Constants/buy-sell';
+import { TAdvertProps } from 'Types/adverts.types';
+import QuickAddModalContent from './quick-add-modal-content';
+import QuickAddModalButtons from './quick-add-modal-buttons';
+
+type TQuickAddModalProps = {
+    advert: Partial<TAdvertProps>;
+};
+
+const QuickAddModal = ({ advert }: TQuickAddModalProps) => {
+    const { my_ads_store, my_profile_store } = useStores();
+    const {
+        should_show_add_payment_method,
+        show_filter_payment_methods,
+        hideQuickAddModal,
+        onClickUpdatePaymentMethods,
+        setSearchTerm,
+        setSearchedResults,
+        setShouldShowAddPaymentMethod,
+        setShowFilterPaymentMethods,
+    } = my_ads_store;
+    const { selected_payment_method, getPaymentMethodsList } = my_profile_store;
+
+    const { is_modal_open, showModal, useSavedState } = useModalManagerContext();
+
+    const type = advert ? advert.type : null;
+
+    const [selected_methods, setSelectedMethods] = useSavedState('selected_methods', []);
+
+    const is_buy_advert = type === buy_sell.BUY;
+    const is_sell_ad_add_payment_methods_selected = !is_buy_advert && selected_payment_method.length > 0;
+    const is_buy_ad_add_payment_methods_selected = is_buy_advert && selected_methods.length > 0;
+    const is_payment_methods_selected =
+        is_sell_ad_add_payment_methods_selected || is_buy_ad_add_payment_methods_selected;
+
+    const setShouldCloseAllModals = (should_close_all_modals: boolean) => {
+        if (show_filter_payment_methods) {
+            setShowFilterPaymentMethods(false);
+            setSearchTerm('');
+            setSearchedResults([]);
+        } else if (is_payment_methods_selected) {
+            localStorage.setItem('selected_methods', JSON.stringify(selected_methods));
+            showModal({
+                key: 'CancelAddPaymentMethodModal',
+                props: {
+                    should_hide_all_modals_on_cancel: is_buy_advert,
+                    onCancel: () => {
+                        my_ads_store.payment_method_ids = [];
+                        my_ads_store.payment_method_names = [];
+                    },
+                },
+            });
+        } else {
+            my_ads_store.payment_method_ids = [];
+            my_ads_store.payment_method_names = [];
+            if (should_close_all_modals) {
+                setShouldShowAddPaymentMethod(false);
+                hideQuickAddModal();
```

Comment:
Why are we doing doing this instead of using hideModal with `should_hide_all_modals` option set to true

Reply:
the function is from my ads store and it already has hideModal in it, but it also calls setSelectedAdId as well, this was pulled from my other PR so its probs better to remove it and just call it straight inside the function itself and remove it from the store 

```
hideQuickAddModal() {
        this.root_store.general_store.hideModal();
        this.setSelectedAdId(undefined);
    }
```

Reply:
oh wait i cant remove this cause its being called in another function in the store... ill just leave this for now, but yeah